e2e/browser/src/components/notifications/index.tsx

```diff
@@ -21,13 +21,127 @@ const MessageList = (props: { messages: Array<any> }) => {
   );
 };
 
+const WebSocketButtons = ({ socket }: { socket?: WebsocketNotification }) => {
+  if (socket === undefined) {
+    return <></>;
+  }
+  return (
+    <>
+      <button
+        onClick={async (e) => {
+          e.preventDefault();
+          await socket.connect();
+        }}
+        data-testid="connectSocket"
```

Comment:
You could also have this change from a connect button to a disconnect button depending on connection state.

Reply:
That's a good idea, I've updated the app slightly to that effect.